app/src/main/java/com/github/orkest/shazam/data/AudioRecording.kt

```diff
@@ -0,0 +1,135 @@
+            require(audioRecord.state == AudioRecord.STATE_INITIALIZED) {
+                Log.d("AudioRecord","AudioRecord is not initialized")
+            }
+
+            //Build the buffer to store the audio chunks from the audioRecord (microphone)
+            val readBuffer = buildReadingBuffer()
+
+            //Return the flow of AudioChunks, will be actually run in a suspending way when collected
+            return flow {
+                audioRecord.startRecording()
+                //Collect audio chunks while the coroutine is active
+                while (currentCoroutineContext().isActive) {
+                    val actualRead = audioRecord.read(readBuffer, 0, readBuffer.size)
+                    val byteArray = readBuffer.sliceArray(0 until actualRead)
+                    val audioChunk = AudioChunk(byteArray, actualRead, System.currentTimeMillis())
+                    emit(audioChunk)
+                }
+                //Stop the audioRecord when the coroutine is cancelled
+                // (either because of result, or user stopped recording)
+                audioRecord.release()
+            }
+        }
+
+        /**
+         * Collects recorded AudioChunks continuously and logs them
+         * @param coroutineScope the same coroutineScope as the streaming session
+         * @throws [SecurityException] if the app does not have the RECORD_AUDIO permission
+         */
+        @RequiresPermission(Manifest.permission.RECORD_AUDIO)
+        fun logsRecordedAudio(coroutineScope: CoroutineScope){
+            coroutineScope.launch {
+                recordingFlow(coroutineScope).collect {
+                    Log.d("HasOnesBuffer", it.buffer.sum().toString())
+                    Log.d("LengthBuffer", it.buffer.size.toString())
+                    Log.d("ShazamActivity", it.timestamp.toString())
+                }
+            }
+
+        }
+
+        /**
+         * Stops the collection of the recorded AudioChunks
+         * Must be called in the same coroutineScope as logsRecordedAudio
+         * @param coroutineScope the same coroutineScope as logsRecordedAudio
+         */
+        fun stopRecording(coroutineScope: CoroutineScope){
+            coroutineScope.cancel()
+        }
+    }
+
+    data class AudioChunk (
+        val buffer: ByteArray = ByteArray(ShazamConstants.SHAZAM_SESSION_READ_BUFFER_SIZE),
+        val meaningfulLengthInBytes: Int = ShazamConstants.SHAZAM_SESSION_READ_BUFFER_SIZE,
+        val timestamp: Long = System.currentTimeMillis()
+    ) {
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
+
+            other as AudioChunk
+
+            return buffer.contentEquals(other.buffer) &&
+                    meaningfulLengthInBytes == other.meaningfulLengthInBytes &&
+                    timestamp == other.timestamp
+        }
+
+        override fun hashCode(): Int {
```

Comment:
Just to clarify, since you will not user Shazam's recognition mechanism, is this the hashCode you will use to compare your audio recordings?

Reply:
No, I will still use Shazam's recognition mechanism *which consists of the generation of a special signature by Shazam, unfortunately the details about that signature generation is not public. The only thing I will not use will be the Shazam catalog and will build my own using Shazam API still since it allows to do that.